pkg/gui/context/search_trait.go

```diff
@@ -23,6 +28,13 @@ func (self *SearchTrait) GetSearchString() string {
 
 func (self *SearchTrait) SetSearchString(searchString string) {
 	self.searchString = searchString
+	if searchString != "" {
```

Comment:
Maybe it would be better just to early return if `searchString` is empty?

No reason to set `self.searchString` if it's empty, and it would be a bit cleaner.

Reply:
I agree that returning early when search string is empty would look cleaner and I included it in a fixup commit. 
2a3bd0991680f3bac8deb9b000dd143af5dd0722

Failing to set `self.searchString` even if `searchString` is empty seems to break at least one of the integration tests: `nested_filter.go`. So I left it as is.